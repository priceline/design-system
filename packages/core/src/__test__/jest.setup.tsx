import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
