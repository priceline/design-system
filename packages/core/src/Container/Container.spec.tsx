import { render } from '../__test__/testing-library'
import { Container } from './Container'

describe('Container', () => {
  test('renders with default theme max-width', () => {
    const { asFragment } = render(<Container />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with maxWidth', () => {
    const { asFragment } = render(<Container maxWidth={1024} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with size', () => {
    const { getByTestId } = render(<Container size='md' data-testid='container' />)

    expect(getByTestId('container')).toHaveStyleRule('max-width', '768px')
  })
})
