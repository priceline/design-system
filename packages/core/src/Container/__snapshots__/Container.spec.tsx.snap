// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders with default theme max-width 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  max-width: 1280px;
}

.c2 {
  margin-left: auto;
  margin-right: auto;
  width: 100%;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    />
  </div>
</DocumentFragment>
`;

exports[`Container renders with maxWidth 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  max-width: 1024px;
}

.c2 {
  margin-left: auto;
  margin-right: auto;
  width: 100%;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    />
  </div>
</DocumentFragment>
`;
